bin/tictactoe

```diff
@@ -1,3 +1,53 @@
 #!/usr/bin/env ruby
 
 require_relative '../config/environment'
+
+class Cli
+
+	def run_game
+		puts "Welcome to Tic-Tac-Toe."
+		puts "Please input your choice (0-2):  0 , 1, or 2 Players."
+
+		choice = gets.chomp
+		case choice
+
+		when "0"
+			Game.new(Computer.new('X'), Computer.new('O')).play
+			
+		when "1"
+			puts "Please would be X, player or computer (P/C):"
+			input = gets.chomp.upcase
+			Game.new(Human.new('X'), Computer.new('O')).play if input == 'P'
+			Game.new(Computer.new('X'), Human.new('O')).play if input == 'C'
+
+		when "2"
+			puts " which player wants to be X (1/2):"
+			input = gets.chomp.to_i
+			if input == 1
+				human_1 = Human.new('X')
+				human_2 = Human.new('O')
+			elsif input == 2
+				human_1 = Human.new('O')
+				human_2 = Human.new('X')
+			end
+
+			Game.new(human_1, human_2).play
+		end
+		puts "Do you want to play again (Y/n)?"
+		input = gets.chomp.upcase
+		if input == "Y"
+			Cli.new.run_game
+		else
+			puts "Thanks for playing TicTacToe. Have a nice day!"
+			exit
+		end
+	end
+
+end
+
+
+Cli.new.run_game
```

Comment:
Prefer to encapsulate CLI logic into a class. For example, the only code in your bin executable should be:

```ruby
#!/usr/bin/env ruby
       
require_relative '../config/environment'

TicTacToeCLI.new.call
```

And then you'd have a file: `lib/tic_tac_toe_cli.rb` with:

```ruby
class TicTacToeCLI
  def initialize
  end

  def call
   # Your game logic
  end
end
```

The code in your bin would generally just go in `call`.